drivers/iommu/riscv/iommu-bits.h

```diff
@@ -1,452 +1,689 @@
 /* SPDX-License-Identifier: GPL-2.0-only */
 /*
- * RISC-V IOMMU Register Layout and Data Structures.
+ * Copyright © 2022-2023 Rivos Inc.
+ * Copyright © 2023 FORTH-ICS/CARV
+ * Copyright © 2023, RISC-V IOMMU Task Group
  *
- * Copyright (C) 2022-2023 Rivos Inc.
+ * RISC-V Ziommu - Register Layout and Data Structures.
  *
- * This program is free software; you can redistribute it and/or modify
- * it under the terms of the GNU General Public License as published by
- * the Free Software Foundation; either version 2 of the License.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
+ * Based on the IOMMU spec version 1.0, 3/2023
+ * https://github.com/riscv-non-isa/riscv-iommu
  *
- * You should have received a copy of the GNU General Public License along
- * with this program; if not, see <http://www.gnu.org/licenses/>.
  */
 
 #ifndef _RISCV_IOMMU_BITS_H_
 #define _RISCV_IOMMU_BITS_H_
 
 #include <linux/types.h>
+#include <linux/bitfield.h>
 #include <linux/bits.h>
-
-/*
- * RISC-V IOMMU register layout and data structures.
- *
- * Based on RISC-V IOMMU Architecture Specification, Version 0.9, 01/2023
- * https://github.com/riscv-non-isa/riscv-iommu
+ 
+/*********************************************\
+* Chapter 5: Memory Mapped register interface *
+\*********************************************/
+
+/* Common field positions */
+#define RISCV_IOMMU_PPN_FIELD		GENMASK(53, 10)
+#define RISCV_IOMMU_QUEUE_SIZE_FIELD	GENMASK(4, 0)	/* As LOG2(size) - 1 */
+#define RISCV_IOMMU_QUEUE_INDEX_FIELD	GENMASK(31, 0)
```

Comment:
As this is not used, what was the reason to define it instead of using run-time (q->cnt - 1)?

Reply:
It's used in the header file for consistency.